### PR TITLE
30 support other llms than openai

### LINKED ISSUE #30: Support other LLMs than OpenAI
The lmos-router implementation currently only supports OpenAI as a model provider.
Arc in comparison has multiple client options, such as Ollama or even more general LangChain4j.

It would be good if the router would also support such options.
Would it make sense to move certain client classes from arc to something like lmos-commons, so that it does not have to be replicated for the router?

### PR DESCRIPTION
**Pull request by xmxnt**
Added Langchain4j as default client which supports multiple language model providers.

### CODE REVIEW

File: lmos-router-llm/build.gradle.kts [hunk 1]
```diff
@@ -5,7 +5,13 @@
 dependencies {
     api(project(":lmos-router-core"))
     implementation("org.slf4j:slf4j-api:1.7.25")
-    implementation("com.azure:azure-ai-openai:1.0.0-beta.10")
-    implementation("com.azure:azure-identity:1.14.0")
     implementation("org.jetbrains.kotlinx:kotlinx-serialization-json-jvm:1.7.3")
+    implementation("dev.langchain4j:langchain4j-open-ai:0.36.2")
```
Comment (RobWin) [inline lmos-router-llm/build.gradle.kts]: Must be compileOnly or?

File: lmos-router-llm/ReadMe.md [hunk 2]
```diff
@@ -89,26 +106,73 @@ The `ExternalModelPromptProvider` class provides a prompt from an external file.
...
 ## Usage
 
 ### Step 1: Initialize the DefaultModelClient
 
 ```kotlin
 val defaultModelClientProperties = DefaultModelClientProperties(
-    openAiApiKey = "your-openai-api-key"
+    openAiUrl = "https://api.openai.com/v1/chat/completions",
+    openAiApiKey = "your-openai-api-key",
+    model = "gpt-4",
```
Comment (kaikreuzer) [inline lmos-router-llm/ReadMe.md]:
```suggestion
    model = "gpt-4o",
```

File: lmos-router-llm/ReadMe.md [hunk 3]
```diff
@@ -131,4 +200,91 @@ val llmAgentRoutingSpecsResolver = LLMAgentRoutingSpecsResolver(
...
+### Advanced: Using LangChainModelClient
+
+For enhanced flexibility and support for multiple language model providers, you can utilize the `LangChainModelClient` along with the `LangChainChatModelFactory`.
+
+```kotlin
+// Define model client properties
+val langChainProperties = ModelClientProperties(
+    provider = LangChainClientProvider.OPENAI.name.lowercase(),
+    apiKey = "your-openai-api-key",
+    model = "gpt-4",
```
Comment (kaikreuzer) [inline lmos-router-llm/ReadMe.md]:
```suggestion
    model = "gpt-4o",
```

File: lmos-router-llm/ReadMe.md [hunk 4]
```diff
@@ -131,4 +200,91 @@ val llmAgentRoutingSpecsResolver = LLMAgentRoutingSpecsResolver(
...
+
+Configure the `DefaultModelClientProperties` or `ModelClientProperties` based on the chosen provider. Ensure that all required fields such as `apiKey`, `model`, and `baseUrl` (if applicable) are correctly set.
+
+**Example Configuration for OpenAI:**
+
+```kotlin
+val properties = ModelClientProperties(
+    provider = LangChainClientProvider.OPENAI.name.lowercase(),
+    apiKey = "your-openai-api-key",
+    model = "gpt-4",
```
Comment (kaikreuzer) [inline lmos-router-llm/ReadMe.md]:
```suggestion
    model = "gpt-4o",
```

File: lmos-router-llm/src/test/kotlin/ai/ancf/lmos/router/llm/LangChainChatModelFactoryTest.kt [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+ */
+class LangChainChatModelFactoryTest {
+    @Test
+    fun `createClient should return OpenAiChatModel for OPENAI provider`() {
+        // Arrange
+        val properties = mockk<ModelClientProperties>()
+        every { properties.provider } returns LangChainClientProvider.OPENAI.name.lowercase()
+        every { properties.baseUrl } returns "https://api.openai.com"
+        every { properties.apiKey } returns "openai-api-key"
+        every { properties.model } returns "gpt-3.5-turbo"
```
Comment (kaikreuzer) [inline lmos-router-llm/src/test/kotlin/ai/ancf/lmos/router/llm/LangChainChatModelFactoryTest.kt]:
```suggestion
        every { properties.model } returns "gpt-4o-mini"
```

**Review by kaikreuzer — DISMISSED** [review]
thanks, lgtm - just some very minor model name corrections as comments below.

**Pull request merged**